### ISSUE TITLE
Pod creation always on pending state for feature- artifacts.

### ISSUE THREAD

**Issue by niljames** [kubeshop/testkube#4386]
I have created a custom executor for running my robot scripts. I have enabled the artifacts feature for my executor. However, when I create and run the tests using the custom executor, the pod creation is always on **pending state**.

But, if I remove the artifacts feature from the executor definition, the tests run just fine, with the pods going through all the stages [init, podinitializing, running, completed and terminating status]. But, I get this error.
![image](https://github.com/kubeshop/testkube/assets/89439509/c466c54c-52af-4a7e-adec-cc750e77cde0)

**To Reproduce**
1. Create a custom executor with the artifacts feature.
2. Create a test with that executor.
3. Run the test.

**Expected behavior**
The test must complete successfully and you should be able to download your artifacts from the artifacts tab.

**Version / Cluster**
- TestKube:
Client Version: 1.14.4
Server Version: 1.14.4
- Kubernetes cluster: AKS
- Kubernetes version: v1.24.6

**Screenshots**
![image](https://github.com/kubeshop/testkube/assets/89439509/f5d9ef23-5192-45ea-b935-f29e71162fee)

**Additional context**
Is there any alternate ways for artifacts setup?

**Comment by vsukhin** [comment]
hey @niljames 
Guess, we discussed this issue a few times in Discord, for example https://discordapp.com/channels/884464549347074049/1150663103588085861
You need to properlt define storage class and mount for your container executor test

**Comment by niljames** [comment]
Forgot to mention this in the bug description.
I had mentioned the storage class and mount as standard and the default volume mount path as seen in the attachment.
![image](https://github.com/kubeshop/testkube/assets/89439509/c6f19ebb-ca94-45a9-b406-941d82d927a8)

**Comment by vsukhin** [comment]
check your pod status and events, it should contain details about the problem like not able to prrovision a volume

**Comment by niljames** [comment]
Yes, I checked the pod status and events. It did contain a binding error with the pvc. I realized that I was using a storage class that did not exist, hence the provisioning failed. The volume was not binding to the pvc. I changed it to the storage class, that I have in my cluster and it worked fine! I was able to save the artifacts in the required directory and also download it.

@vsukhin , Thank you so much for your timely response! Really appreciate it!
Will be closing this issue.
